hack/app_sre_pr_check.sh

```diff
@@ -1,7 +1,32 @@
 #!/bin/bash
 
-set -ex
+#set -ex
 
 cd $(dirname $0)/..
+CONTAINER_ENGINE=$(which docker 2>/dev/null || which podman)
+#convert all metadata.yaml to json first
+yfiles=$(find . -name 'metadata.yaml')
+for f in $yfiles
+do
+   echo "convert $f to json"
+   if ! python3 hack/yamltojson.py $f; then
+     echo "convert failed"
+     exit 1
+   else
+     echo "convert succeed"
+   fi
+done
+#Verify the metadata.json are valide
+jfiles=$(find . -name 'metadata.json')
+for f in $jfiles
+do
+   echo "validating the jsonschema for $f"
+   if ! $CONTAINER_ENGINE run --rm -v $(pwd):/json quay.io/haowang/jsonschema:latest -i /json/$f /json/hack/metadata.schema.json; then
```

Comment:
can we invoke a single container and then run a `find ... exec ...` in there? I think it's better than invoking docker each time

Reply:
This is more easy for me ,we can fail at anytime when the validation failed.